crates/polars-python/src/conversion/any_value.rs

```diff
@@ -473,16 +517,23 @@ pub(crate) fn py_object_to_any_value<'py>(
         }
     }
 
-    let type_object_ptr = ob.get_type().as_type_ptr() as usize;
+    let py_type = ob.get_type();
+    let py_type_address = py_type.as_ptr() as usize;
+
+    Python::with_gil(move |py| {
+        LUT.with_gil(py, move |lut| {
+            if !lut.contains_key(&py_type_address) {
+                let k = TypeObjectKey::new(py_type.clone().unbind());
+
+                assert_eq!(k.address, py_type_address);
 
-    Python::with_gil(|py| {
-        let conversion_function = get_conversion_function(ob, py, allow_object)?;
```

Comment:
Fix a regression from https://github.com/pola-rs/polars/commit/ebd53027d7aeea5db270b23b4a73999668c6ad08#diff-778ed52e7b4a402304a62e51974c92208369baf61c50ad78eca715b7f6f18d89R478-R485 that caused us to always call `get_conversion_function` regardless of whether it existed in the cache.